File: javascript/difference-of-squares/difference-of-squares.js
export default class Square {
  constructor(num) {
    this.num = num
    this.n = Array.from(Array(this.num), (e, i) => i + 1)
  }


  get difference() {
    return this.squareOfSums - this.sumOfSquares
  }

  get sumOfSquares() {
   return this.n.reduce((acc, curr) => acc + Math.pow(curr, 2)) 
  }

  get squareOfSums() {
    const y =  this.n.reduce((acc, curr) => acc + curr)
    return Math.pow(y, 2)
  }
}

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(num)
    @num = num
  end

  def square_of_sum 
    sum = 0
    for i in 1..@num
      sum += i
    end
    sum ** 2
  end
  
  def sum_of_squares 
    square = []
    for i in 1..@num
      square << i ** 2
    end
   square.inject(:+)  
  end

  def difference
    square_of_sum - sum_of_squares
  end
end
